ghtt/assignment.py

```diff
@@ -287,12 +306,12 @@ def create_repos(ctx, source, yes, students=None, groups=None):
         subprocess.check_call(["git", "add", "-A"], cwd=source)
         subprocess.call(["git", "commit", "-m", "fill in templates"], cwd=source)
         click.secho("Pushing source to {}".format(g_repo.ssh_url), fg="green")
-        subprocess.check_call(["git", "push", g_repo.ssh_url, "{}:master".format(repo.name)], cwd=source)
-        subprocess.check_call(["git", "checkout", "master"], cwd=source)
+        subprocess.check_call(["git", "push", g_repo.ssh_url, f"{repo.name}:{g_repo.master_branch or default_branch}"], cwd=source)
```

Comment:
Always push to the specified default branch, even if repo is different. This makes `ghtt` behavior more predictable.

```suggestion
        subprocess.check_call(["git", "push", g_repo.ssh_url, f"{repo.name}:{default_branch}"], cwd=source)
```